cli/cmd/login.go

```diff
@@ -97,10 +98,19 @@ func runLogin(ctx context.Context, cmd *cobra.Command) (err error) {
 	mux := http.NewServeMux()
 	refreshToken := ""
 	gotToken := make(chan struct{})
+	var callbackLock gosync.Mutex
+	callbackHandled := false
 	mux.HandleFunc("/callback", func(w http.ResponseWriter, r *http.Request) {
+		callbackLock.Lock()
```

Comment:
Should use `sync.Once` which is for this type of thing.

Reply:
Also we still want to do the redirect to `success-close`

Reply:
Ok so https://github.com/cloudquery/cloudquery/pull/17331/commits/f1b1807890f84c0dfd5d868ffa03575c7ccf8527 looks better indeed